arch/common/fill_with_zero.ld

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2023, Google, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * LLVM LLD fills empty spaces (created using ALIGN() or moving the location
+ * counter) in executable segments with TrapInstr pattern, e.g. for ARM the
+ * TrapInstr pattern is 0xd4d4d4d4. GNU LD fills empty spaces with 0x00
+ * pattern.
+ *
+ * We may want to have some section (e.g. rom_start) filled with 0x00,
+ * e.g. because MCU can interpret the pattern as a configuration data.
```

Comment:
Isn't the root cause here that we have a linker script somewhere putting interpretable data in what the linker thinks is a .text section?  That's likely to cause other problems too.  No complaints about the patch per se, but this smells like something needs to be fixed elsewhere.

Reply:
Hmm...actually I don't know what is the original problem :smile:, @nordicjm found that when ROM_START_OFFSET != 0 then the empty space is filed with `0xd4d4d4d4` when compiled with LLVM. GNU LD fills it with 0x00.

My opinion is that the pattern used to fill empty spaces shouldn't cause any problems, because these spaces are generally not used, but there are MCUs which need configuration data at the beginning of the flash, so `rom_start` section is appropriate for this task.

Reply:
LLD fills empty space with what they call `TrapInstr` https://reviews.llvm.org/D33630. For ARM `TrapInstr` is `0xd4d4d4d4`.